File: ReminderSystem.cpp
#include "ReminderSystem.hpp"
#include <algorithm>


void ReminderSystem::setReminder(const Reminder& reminder) {
    reminders.push_back(reminder);
}


void ReminderSystem::triggerNotifications() {
}


void ReminderSystem::cancelReminder(const std::string& id) {
    reminders.erase(std::remove_if(reminders.begin(), reminders.end(), [&](const Reminder& r) { return r.id == id; }), reminders.end());
}

File: FitnessAppCsharp/ReminderSystem.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace FitnessApp
{
    public class ReminderSystem : INotifiable
    {
        private string userId;
        private List<Reminder> reminders = new List<Reminder>();
        private NotificationType notificationType;

        public void SetReminder(Reminder reminder)
        {
            reminders.Add(reminder);
        }

        public void TriggerNotifications()
        {
            // Заглушка
        }

        public void CancelReminder(string id)
        {
            reminders.RemoveAll(r => r.Id == id);
        }

        public void SendNotification(string message)
        {
            Console.WriteLine($"Reminder notification: {message}");
        }
    }
}